### PR TITLE
feat(table): fixed first column

### PR DESCRIPTION
**Pull request by malegreIndec**
# Description

Add new variant to the table component

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How to test

1. List instructions for how to test
1. Make sure it's detailed and includes the exact steps

### CODE REVIEW

File: src/theme/components/table.js [hunk 1]
```diff
@@ -29,6 +38,28 @@ const table = {
...
+                    backgroundColor: 'brand.neutral50'
+                },
+                '&:nth-of-type(odd) td:first-of-type ': {
+                    backgroundColor: 'brand.white'
+                }
+            },
+
+            td: {
+                '&:first-of-type': {
+                    ...stickyTable
```
Comment (maximilianoforlenza) [inline src/theme/components/table.js:60]: aca no estas extendiendo este objeto y agregando nuevas props, por lo cual directamente podes hacer esto

```js
'&:first-of-type': stickyTable
```

File: src/theme/components/table.js [hunk 2]
```diff
@@ -29,6 +39,27 @@ const table = {
                 textAlign: 'left',
                 padding: '10px 20px'
             }
+        },
+
+        fixedFirstColumn: {
+            ...baseStyles,
+            tr: {
+                '& th:first-of-type': {
```
Comment (arielerv) [inline src/theme/components/table.js:47]:
```
'& th:first-of-type': fixedFirstColumnStyles,
'&:nth-of-type(even) td:first-of-type': {
```

File: src/theme/components/table.js [hunk 3]
```diff
@@ -29,6 +39,27 @@ const table = {
                 textAlign: 'left',
                 padding: '10px 20px'
             }
+        },
+
```
Comment (arielerv) [inline src/theme/components/table.js:43]: borrar espacio

File: src/theme/components/table.js [hunk 4]
```diff
@@ -29,6 +39,27 @@ const table = {
...
+                    ...fixedFirstColumnStyles
+                },
+                '&:nth-of-type(even) td:first-of-type ': {
+                    backgroundColor: 'brand.neutral50'
+                },
+                '&:nth-of-type(odd) td:first-of-type ': {
+                    backgroundColor: 'brand.white'
+                }
+            },
+
```
Comment (arielerv) [inline src/theme/components/table.js:57]: borrar espacio

File: src/theme/components/table.js [hunk 5]
```diff
@@ -29,6 +39,27 @@ const table = {
...
+            td: {
+                '&:first-of-type': {
```
Comment (arielerv) [inline src/theme/components/table.js:59]:
```
'&:first-of-type': fixedFirstColumnStyles
```

File: src/theme/components/table.js [hunk 6]
```diff
@@ -29,6 +39,27 @@ const table = {
...
             }
+        },
+
+        fixedFirstColumn: {
+            ...baseStyles,
+            tr: {
+                '& th:first-of-type': {
+                    ...fixedFirstColumnStyles
+                },
+                '&:nth-of-type(even) td:first-of-type ': {
```
Comment (arielerv) [inline src/theme/components/table.js:50]: borrar espacio al final

File: src/theme/components/table.js [hunk 7]
```diff
@@ -29,6 +39,27 @@ const table = {
...
+                    backgroundColor: 'brand.neutral50'
+                },
+                '&:nth-of-type(odd) td:first-of-type ': {
```
Comment (arielerv) [inline src/theme/components/table.js:53]: borrar espacio al final

**Review by arielerv — APPROVED** [review]

**Pull request merged**